Task12

evgenum opened the pull request.

Commit:
Ответ от контроллера вынесен в отдельный класс.

Commit:
Методы вынесены в отдельные классы

vatt2001 commented on src/main/scala/lectures/oop/FatUglyController.scala:
```diff
@@ -44,35 +46,40 @@ class FatUglyController {
   }
+}
 
+object Mq {
```
Объекты - это хорошо, но их совершенно невозможно тестировать отдельно от контроллера. Давай выделим интерфейс, от которого будет зависеть контроллер, а этот код сделаем реализацией интерфейса. И его можно будет тестировать отдельно и независимо.

vatt2001 commented on src/main/scala/lectures/oop/FatUglyController.scala:
```diff
@@ -24,16 +23,19 @@ import java.security.MessageDigest
+    val databaseConnectionId = Database.connectToPostgresDatabase()
+    val mqConnectionId = Mq.connectToIbmMq()
+    LocalMailer.initializeLocalMailer()
+    if (route == uploadRoute) {
```
Для матчинга роутов отлично подходит pattern matching

vatt2001 commented on src/main/scala/lectures/oop/FatUglyController.scala:
```diff
@@ -24,16 +23,19 @@ import java.security.MessageDigest
+      } else if (requestBody.get.length > maxRequestLength) {
+        ForbiddenFileSize
       } else {
         val stringBody = new String(requestBody.get.filter(_ != '\r'))
```
Обработка этого роута выглядит слишком сложно и является отличным претендентом на вынос в отдельный метод

vatt2001 replied:
Аналогично - и обработка каждого файла

Commit:
Рефакторинг.
Для обработки условий используется pattern matching.
Обработка запроса и файла вынесена в отдельный метод.
Объекты внешних ресурсов отделены от конроллера интерфейсами.
